src/lib/server/functions/getGraphData.ts

```diff
@@ -0,0 +1,86 @@
+import prisma from "$lib/server/database.js";
+import type { GraphMetrics } from "@prisma/client";
+import type { ChartConfiguration } from "chart.js";
+import dayjs from "dayjs";
+import { inPlaceSort } from "fast-sort";
+
+export default async function getGraphData(category: string, user?: string) {
+  let userItemCount: GraphMetrics[] = [];
+
+  if (user && user.match(/^\d{17,19}$/)) {
+    const privacyCheck = await prisma.preferences.findUnique({
+      where: { userId: user },
+      select: { leaderboards: true },
+    });
+
+    if (!privacyCheck) return "invalid user";
+
+    if (privacyCheck?.leaderboards) {
+      userItemCount = await prisma.graphMetrics.findMany({
+        where: {
+          AND: [
+            { userId: user },
+            { date: { gte: dayjs().subtract(45, "days").toDate() } },
+            { category: category },
+          ],
+        },
+      });
+    } else return "private profile"
+  } else return "invalid user";
```

Comment:
Can clean this up by doing the ‘invalid’ checks first

Reply:
cleaned